assets/js/modules/analytics-4/datastore/custom-dimensions.js

```diff
@@ -326,12 +400,27 @@ const baseSelectors = {
 	 * @return {boolean} TRUE if the available custom dimensions are being synced, otherwise FALSE.
 	 */
 	isSyncingAvailableCustomDimensions: createRegistrySelector(
-		( select ) => () => {
-			return select(
-				MODULES_ANALYTICS_4
-			).isFetchingSyncAvailableCustomDimensions();
+		( select ) => ( state ) => {
+			return (
+				select(
+					MODULES_ANALYTICS_4
+				).isFetchingSyncAvailableCustomDimensions() ||
+				!! state?.syncTimeoutID
```

Comment:
I'm not sure we need to reference the timeout ID here, do we? I think isSyncing can probably remain as it was 🤔

Reply:
I've removed the selector entirely (and updated its usage) as it just uses `isFetchingSyncAvailableCustomDimensions` under the hood. Thanks!

Reply:
I actually added it because it causes a 2 second (or more for multiple tiles) flicker between the loading state and the error state when the report request fails due to an invalid custom dimensions error and needs to re-sync. Thoughts?